kuri_camera/README.md

```diff
@@ -1,17 +1,44 @@
-certain madmux functions. Use the Python scripts for now.
\ No newline at end of file
+### C++
+
+- NOTE: For the C++ code to compile, you have to manually link to `libmadmux` and multiple other libraries. This requires finding the paths to these libraries. The CMake file contains the paths to those libraries on **our** computers -- use those as pointers to find those libraries on **your** computers, andthen change the paths accordingly.
+- `ros_publisher` is the second-simplest and second-slowest node to use. This does the same thing as `ros_publisher.py`, except in C++.
+- The fastest way to receive images is to use H264 compression (which compresses the image in space and time, as opposed to the `ros_publisher`s which only compress them in space). Our approach involves porting the Unix Domain Socket (UDS) data stream published by Kuri's camera directly to a TCP socket, and then reading from that TCP socket on a remote computer and processing the images.
+  - `uds_to_TCP` is the node that reads bytes from the UDS socket and ports them over to a TCP socket. This must be run **on the Kuri**.
+  - `tcp_test_client` is just a test client you can run to ensure that data is getting streamed to the TCP socket as expected. It will partition the stream into H264 frames and print the size and beginning bytes of each frame.
+  - `h264_decoder_node` subscribes to the TCP socket, decodes the H264 stream, and converts the image to a cv::Mat. The current code then displays it using OpenCV's imshow and publishes it as a ros Image msg -- however, there is a sample function demonstrating how to use the cv::Mat for arbitrary processing, while remaining thread-safe. This node should be run **on a remote computer**. (NOTE: if you would like to use the H264 stream but process images on-board the Kuri, we recommend you change `h264_decoder.cpp` to read directly from the UDS socket, instead of the TCP socket that is generated by `uds_to_TCP`).
+- NOTE: you must specify the tcpSocketHostname and the tcpSocketPort. This can be done via rosparams.
+
+### Miscellaneous
+- `display_compressed_images` will read a CompressedImage rosmsg, display it, and print out the delay between when the message was received and the timestamp on the message. This can be used in conjunction with the `ros_publisher` nodes.
+
+## Latency Analysis
+- To determine **end-to-end latency**, we set up a computer near our remote computer and ran a stopwatch on it. This was our ground truth. We then pointed the Kuri's camera at this computer and used the above node(s) to display Kuri's image on the remote computer. We then took a video of the ground truth computer and the adjacent remote computer. We then played bacl the video and paused it at certain frames. The difference is the end-to-end latency. We have seen it as low as 300-400 ms when just displaying the cv::Mat (it goes to 400-500 ms when also publishing a ros Image msg).
```

Comment:
```suggestion
- To determine **end-to-end latency**, we set up a computer near our remote computer and ran a stopwatch on it. This was our ground truth. We then pointed the Kuri's camera at this computer and used the above node(s) to display Kuri's image on the remote computer. We then took a video of the ground truth computer and the adjacent remote computer. We then played back the video and paused it at certain frames. The difference is the end-to-end latency. We have seen it as low as 300-400 ms when just displaying the cv::Mat (it goes to 400-500 ms when also publishing a ros Image msg).
```

Reply:
Am including this change in the larger commit addressing your comments.